### PR TITLE
Use the same icon in all notifications

### LINKED ISSUE #4995: Foreground service notifications should use the same notification
Our foreground services currently have different notification icons - audio recording uses a specific recording one and location tracking uses the same icon as our other notifications (`R.drawable.ic_notification_small`). These should be made consistent, so it's clear to users when Collect is up to something in the background - they should all use `R.drawable.ic_notification_small`.

### Notes

We'll need a way to share the icon between multiple packages for this to work. It could be that using a "provider" interface that the `Application` can implement (similarly to `LocalizedApplication`) would solve this.

### PR DESCRIPTION
**Pull request by grzesiek2010**
Closes #4995 

#### What has been done to verify that this works as intended?
I've verified the fix manually.

#### Why is this the best possible solution? Were any other approaches considered?
The only thing of note here is the way we share the icon used in notifications across different modules. I decided to just place that icon in the `androidshared` module. I like this option more than for example initializing modules and injecting that icon because if we decided to chose the second option we would need to do that for other resources to be consistent (like for `ic_map_point` for example). In the future we would end up with bigger and bigger providers/initializers.

#### How does this change affect users? Describe intentional changes to behavior and behavior that could have accidentally been affected by code changes. In other words, what are the regression risks?
It's a safe fix that should have no side effects. Please just verify that all notifications use the same icon.

#### Do we need any specific form for testing your changes? If so, please attach one.
Any form with the [Audio Widget](https://docs.getodk.org/form-question-types/#audio-widgets).

#### Does this change require updates to documentation? If so, please file an issue [here]( https://github.com/getodk/docs/issues/new) and include the link below.
No.

#### Before submitting this PR, please make sure you have:
- [x] run `./gradlew checkAll` and confirmed all checks still pass OR confirm CircleCI build passes and run `./gradlew connectedDebugAndroidTest` locally.
- [x] verified that any code or assets from external sources are properly credited in comments and/or in the [about file](https://github.com/getodk/collect/blob/master/collect_app/src/main/assets/open_source_licenses.html).
- [x] verified that any new UI elements use theme colors. [UI Components Style guidelines](https://github.com/getodk/collect/blob/master/docs/CODE-GUIDELINES.md#ui-components-style-guidelines)

### CODE REVIEW

**Review by seadowg — CHANGES_REQUESTED** [review]
I'd like to try and avoid these kinds of static dependencies as they prevent they pollute these modules with "Collect" specific links that would make it hard for another app to use the module (not essential, but I think a good standard to try and adhere to in terms of the levels of separation we have in our modules). However, our reliance on `strings` already breaks that. If we wanted that kind of separation, we'd also need a dynamic/inverted way of grabbing strings.

Just to keep the `androidshared` a little cleaner, could we introduce a `icons` module that we use to share icons between modules (like we do with `strings`)? That at least makes it very clear when a module needs access to external (to the module itself) icons.

File: settings.gradle [hunk 1]
```diff
@@ -26,8 +26,9 @@ include ':settings'
 include ':servicetest'
 include ':maps'
 include ':osmdroid'
+include ':icons'
 
 apply from: 'secrets.gradle'
 if (getSecrets().getProperty('MAPBOX_DOWNLOADS_TOKEN', '') != '') {
     include ':mapbox'
-}
+}
```
Comment (seadowg) [inline settings.gradle]: Looks like there's a missing new line at the end of the file here.

**Review by seadowg — APPROVED** [review]

**Comment by grzesiek2010** [comment]
> Just to keep the androidshared a little cleaner, could we introduce a icons module that we use to share icons between modules (like we do with strings)? That at least makes it very clear when a module needs access to external (to the module itself) icons.

This should be ready. I will file a separate issue for moving other icons too.

File: settings.gradle (hunk 1 above)
Comment (grzesiek2010) [reply settings.gradle]: Why do we need that line?

Comment (seadowg) [reply settings.gradle]: Git (and Github) warns against it (there's a good SO on why [here](https://stackoverflow.com/questions/5813311/whats-the-significance-of-the-no-newline-at-end-of-file-log)). You can set up Android Studio to make sure you always have a new line in its settings (Editor > General > On Save > Ensure every saved file ends with a line break).

Comment (grzesiek2010) [reply settings.gradle]: Ok done.

**Comment by dbemke** [comment]
**Tested with Success!**
Verified on Androids: 8.1, 11

**Verified icons in notifications:**

- for audio recording in Audio widget (All widgets form)
- for background audio recording
- for internal audio recording with other questions
- for location recording in Geopoint, Geoshape/Geotrace with automatic location recording
- form update failed/succeeded
- form upload succeeded
- when match exactly is enabled and disabled

**Comment by srujner** [comment]
**Tested with Success!**
Verified also on Androids: 5.1, 10 and 12

**Pull request merged**